aws-lc-sys/builder/main.rs

```diff
@@ -314,13 +314,39 @@ trait Builder {
     fn build(&self) -> Result<(), String>;
 }
 
+#[derive(Clone, Copy, Debug, PartialEq, Eq)]
+pub(crate) enum CStdRequested {
+    C99,
+    C11,
+    None,
+}
+
+impl CStdRequested {
+    fn from_env() -> Self {
+        if let Some(val) = option_env("AWS_LC_SYS_C_STD") {
+            let cstd = match val.as_str() {
+                "99" => CStdRequested::C99,
+                "11" => CStdRequested::C11,
+                _ => CStdRequested::None,
```

Comment:
If it's something we can't parse should this be a hard error? If someone set `AWS_LC_SYS_C_STD=c11` it would hit none here, and then could they actually get c99, or vice versa.

Reply:
Yeah, I thought about matching against a RegExp here to add more flexibility, but I'm not sure it's worth the trouble.

I see this environment variable as something that will rarely be needed. We will default to C11 for most consumers (which should work for any build host that's been setup sometime in the last decade).